CODINGSTYLE.md

```diff
@@ -0,0 +1,335 @@
+# Coding Style Guide
+
+The Scala.js project has pretty drastic rules in terms of coding style.
+This document tries to document them as much as possible.
+
+A *few* of these rules are checked automatically using Scalastyle, but most of them are too complex to teach to an automated tool.
+
+In some cases, [@sjrd](https://github.com/sjrd) will just say something else on a code review.
+His opinion on a direct piece of code always takes precedence over this document.
+Feel free to point out the conflict, though: the document can be improved.
```

Comment:
Same as above, something like

All contributions should follow these guidelines, although you should first and foremost follow eventual suggestions that the core team might have during code review.
Feel free to point out if something they are saying conflicts with this document, we're always looking to improve it!